### PR TITLE
Add files for REST API project

### PR DESCRIPTION
**Pull request by cybertuck**
Initial commit of planning poker API

### CODE REVIEW

File: pokerplanningapi.iml [hunk 1]
```diff
@@ -0,0 +1,127 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (andrii-vv) [inline pokerplanningapi.iml:1]: file must be added to .gitignore and not committed

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.7.3</version>
+        <relativePath/> <!-- lookup parent from repository -->
+    </parent>
+    <groupId>com.sonicstudio</groupId>
+    <artifactId>pokerplanningapi</artifactId>
```
Comment (andrii-vv) [inline pom.xml:12]: you can use '-' for such naming to separate words

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                        <exclude>
+                            <groupId>org.projectlombok</groupId>
+                            <artifactId>lombok</artifactId>
+                        </exclude>
+                    </excludes>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
```
Comment (andrii-vv) [inline pom.xml:81]: add maven-checkstyle-plugin - it will make your project much cleaner

https://www.baeldung.com/checkstyle-java

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-devtools</artifactId>
+            <scope>runtime</scope>
+            <optional>true</optional>
+        </dependency>
+<!--        <dependency>
```
Comment (andrii-vv) [inline pom.xml:42]: what's the reason to keep it?

File: src/main/java/com/sonicstudio/pokerplanningapi/PokerplanningapiApplication.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.context.annotation.Bean;
+
+@SpringBootApplication
+public class PokerplanningapiApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(PokerplanningapiApplication.class, args);
+    }
+
+    @Bean
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/PokerplanningapiApplication.java:14]: potentially can be moved to separate config class under configuration package

File: src/main/java/com/sonicstudio/pokerplanningapi/model/Room.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Set;
+
+@Getter
+@Setter
+@Entity
+@Table(name = "room")
+@ToString
+//@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/model/Room.java:19]: still needed?

File: src/main/java/com/sonicstudio/pokerplanningapi/model/Room.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "room_number")
+    private String room_number;
+
+    @OneToMany(mappedBy = "room", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
+    private Set<User> participants = new HashSet<>();
+
+    @Column(name = "scrummaster")
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/model/Room.java:42]: could be separated with '_'

File: src/main/java/com/sonicstudio/pokerplanningapi/repository/RoomRepository.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.stereotype.Repository;
+
+import java.util.Optional;
+
+@Repository
+public interface RoomRepository extends JpaRepository<Room, Long> {
+
+    Optional<Room> findRoomByLifetime(Integer lifetime);
+
+    Optional<User> findByParticipantsIsNotNull();
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/repository/RoomRepository.java:15]: IsNotNull ? what does it do?

File: src/main/java/com/sonicstudio/pokerplanningapi/model/Room.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        this.participants = new HashSet<>();
+    }
+
+    @Id
+    @Column(name = "id")
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "room_number")
+    private String room_number;
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/model/Room.java:37]: room NAMBER but String?

File: src/main/java/com/sonicstudio/pokerplanningapi/model/User.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    private Long id;
+
+    @Column(name = "uid")
+    private String uid;
+
+    @Column(name = "name")
+    private String name;
+
+    @Column(name = "created")
+    private LocalDateTime created;
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/model/User.java:31]: it has different naming in Room model, such as "created_date"

File: src/main/java/com/sonicstudio/pokerplanningapi/model/User.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+@ToString
+public class User implements Serializable {
+
+    @Id
+    @Column(name = "id", unique = true)
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(name = "uid")
+    private String uid;
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/model/User.java:25]: what's the purpose of having "uid"? is it expected to be not unique?

you already have "id"

File: src/main/java/com/sonicstudio/pokerplanningapi/rest/RoomRestController.java [hunk 12]
```diff
@@ -0,0 +1,106 @@
+package com.sonicstudio.pokerplanningapi.rest;
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/rest/RoomRestController.java:1]: package might be renamed to 'controller'

File: src/main/java/com/sonicstudio/pokerplanningapi/rest/RoomRestController.java [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.persistence.Table;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1/rooms")
+public class RoomRestController {
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/rest/RoomRestController.java:20]: 'Rest' part is optional unless you want to add other types

File: src/main/java/com/sonicstudio/pokerplanningapi/rest/RoomRestController.java [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+
+    @Autowired
+    private RoomService roomService;
+
+    @Autowired
+    private UserService userService;
+
+    @RequestMapping(value = "{id}", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/rest/RoomRestController.java:28]: can be simplified to @GetMapping("{id}", produces = MediaType.APPLICATION_JSON_VALUE) here in in other places

see https://www.baeldung.com/spring-new-requestmapping-shortcuts

File: src/main/java/com/sonicstudio/pokerplanningapi/rest/RoomRestController.java [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+import com.sonicstudio.pokerplanningapi.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.persistence.Table;
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/rest/RoomRestController.java:15]: unused

File: src/main/java/com/sonicstudio/pokerplanningapi/rest/RoomRestController.java [hunk 16]
```diff
@@ -0,0 +1,106 @@
...
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1/rooms")
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/rest/RoomRestController.java:19]: "/api" can be omitted or be more specific, provide service purpose for example

File: src/main/java/com/sonicstudio/pokerplanningapi/service/RoomServiceImpl.java [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+@Slf4j
+@Service
+public class RoomServiceImpl implements RoomService {
+
+    @Autowired
+    RoomRepository roomRepository;
+
+    @Override
+    public Room getById(Long id) {
+        log.info("IN RoomServiceImpl getById {}", id);
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/service/RoomServiceImpl.java:25]: "IN RoomServiceImpl" can be omitted

File: src/main/java/com/sonicstudio/pokerplanningapi/rest/UserRestController.java [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+        return new ResponseEntity<>(user, headers, HttpStatus.CREATED);
+    }
+
+    @RequestMapping(value = "", method = RequestMethod.PUT, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<User> updateUser(@RequestBody @Validated User user){
+        HttpHeaders headers = new HttpHeaders();
+        if(user == null) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+        this.userService.create(user);
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/rest/UserRestController.java:70]: shouldn't it be 'update' instead?

File: src/main/java/com/sonicstudio/pokerplanningapi/service/UserServiceImpl.java [hunk 19]
```diff
@@ -0,0 +1,58 @@
+package com.sonicstudio.pokerplanningapi.service;
+
+import com.sonicstudio.pokerplanningapi.model.Room;
+import com.sonicstudio.pokerplanningapi.model.User;
+import com.sonicstudio.pokerplanningapi.repository.UserRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
```
Comment (andrii-vv) [inline src/main/java/com/sonicstudio/pokerplanningapi/service/UserServiceImpl.java:9]: not used

File: src/test/java/com/sonicstudio/pokerplanningapi/room/RoomServiceImplTest.java [hunk 20]
```diff
@@ -0,0 +1,48 @@
+package com.sonicstudio.pokerplanningapi.room;
```
Comment (andrii-vv) [inline src/test/java/com/sonicstudio/pokerplanningapi/room/RoomServiceImplTest.java:1]: incorrect package

File: src/test/java/com/sonicstudio/pokerplanningapi/room/UserInRoomServiceImplTest.java [hunk 21]
```diff
@@ -0,0 +1,62 @@
+package com.sonicstudio.pokerplanningapi.room;
```
Comment (andrii-vv) [inline src/test/java/com/sonicstudio/pokerplanningapi/room/UserInRoomServiceImplTest.java:1]: incorrect package

File: src/test/java/com/sonicstudio/pokerplanningapi/room/UserServiceImplTest.java [hunk 22]
```diff
@@ -0,0 +1,44 @@
+package com.sonicstudio.pokerplanningapi.room;
```
Comment (andrii-vv) [inline src/test/java/com/sonicstudio/pokerplanningapi/room/UserServiceImplTest.java:1]: incorrect package

File: src/test/java/com/sonicstudio/pokerplanningapi/room/RoomServiceImplTest.java [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+@SpringBootTest
+public class RoomServiceImplTest {
+
+    @Autowired
+    private RoomService roomService;
+
+    @Autowired
+    private RoomRepository roomRepository;
+
+    static final String ROOM_NAME = "142";
```
Comment (andrii-vv) [inline src/test/java/com/sonicstudio/pokerplanningapi/room/RoomServiceImplTest.java:22]: these two might be local to test

File: src/test/java/com/sonicstudio/pokerplanningapi/room/UserServiceImplTest.java [hunk 24]
```diff
@@ -0,0 +1,44 @@
...
+import com.sonicstudio.pokerplanningapi.repository.UserRepository;
+import com.sonicstudio.pokerplanningapi.service.UserService;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SpringBootTest
+public class UserServiceImplTest {
```
Comment (andrii-vv) [inline src/test/java/com/sonicstudio/pokerplanningapi/room/UserServiceImplTest.java:13]: overall, you can add more unit tests to cover positive and negative scenarios for every service.

tests can be added to test both controllers.

**Comment by cybertuck** [comment]
test comment

**Comment by andrii-vv** [comment]
OK

**Review by andrii-vv — CHANGES_REQUESTED** [review]
